purchase_lot/tests/test_purchase_lot.py

```diff
@@ -88,3 +108,48 @@ def test_purchase_lot(self):
         self.assertEqual(pol1.lot_id.id, lot1.id)
         pol1.order_id.button_confirm()
         self.assertEqual(pol1.move_ids.restrict_lot_id.id, lot1.id)
+
+    def test_lot_propagation(self):
+        # Required for `route_id` to be visible in the view
+        self.env.user.groups_id += self.env.ref("stock.group_adv_location")
+
+        # Create a sales order with a line of 200 PCE incoming shipment,
+        # with route_id drop shipping
+        so_form = Form(self.env["sale.order"])
+        so_form.partner_id = self.customer
+        so_form.payment_term_id = self.env.ref(
+            "account.account_payment_term_end_following_month"
+        )
+        with mute_logger("odoo.tests.common.onchange"):
+            # otherwise complains that there's not enough inventory and
+            # apparently that's normal according to @jco and @sle
+            with so_form.order_line.new() as line:
+                line.product_id = self.external_serial_product
+                line.product_uom_qty = 200
+                line.price_unit = 1.00
+                line.route_id = self.env.ref("purchase_stock.route_warehouse0_buy")
+
+        sale_order_drp_shpng = so_form.save()
+        sale_order_drp_shpng.order_line.lot_id = self.env["stock.lot"].create(
+            {
+                "name": "Seq test DS pdt",
+                "product_id": self.external_serial_product.id,
+            }
+        )
+        initial_lot = sale_order_drp_shpng.order_line.lot_id
+        # Confirm sales order
+        sale_order_drp_shpng.action_confirm()
```

Comment:
It is confusing to speak about dropshipping here. 
```suggestion
        # Create a sales order with a line of 200 PCE incoming shipment,
        # with route_id buy
        so_form = Form(self.env["sale.order"])
        so_form.partner_id = self.customer
        so_form.payment_term_id = self.env.ref(
            "account.account_payment_term_end_following_month"
        )
        with mute_logger("odoo.tests.common.onchange"):
            # otherwise complains that there's not enough inventory and
            # apparently that's normal according to @jco and @sle
            with so_form.order_line.new() as line:
                line.product_id = self.external_serial_product
                line.product_uom_qty = 200
                line.price_unit = 1.00
                line.route_id = self.env.ref("purchase_stock.route_warehouse0_buy")

        sale_order = so_form.save()
        sale_order.order_line.lot_id = self.env["stock.lot"].create(
            {
                "name": "Seq test DS pdt",
                "product_id": self.external_serial_product.id,
            }
        )
        initial_lot = sale_order.order_line.lot_id
        # Confirm sales order
        sale_order.action_confirm()
```